src/App.vue

```diff
@@ -424,6 +487,197 @@ export default {
+     * to keep variable names short and readable.
+     *
+     * This method merges the user's and org's settings.
+     * This is needed to show both org level and user level settings in one settings menu.
+     * The global default settings object is used as a structure of the keys to iterate on.
+     * While iterating on the keys of the global default settings object, these rules are followed to merge
+     * - If a key is not present in org's settings, use the key from user's settings and skip to next key
+     * - If a key is present in org's settings, use the scope for that key and move to it's children
+     * - The above process is done for headers, tabs and atomic settings.
+     * - For the lowest level keys (atomic settings), use org's setting value if available otherwise use user's setting value
+     *
+     * @param {Object} userSettings - User's version of settings
+     * @param {Object} orgSettings - Org's version of settings
+     * @returns {Object} An object with user's and org's settings merged (with org's settings taking priority)
+     */
+    mergeSettings(userSettings, orgSettings) {
+      // Making a deep clone of global default settings.
+      // The keys/values will be removed/updated according to user/org settings as we iterate
+      let mergedSettings = clonedeep(globalDefaultSettings);
+
+      for (let [headerName, headerDetails] of Object.entries(mergedSettings)) {
+        // iterating on headers
+        let path = headerName;
+        let headersInOrgSettings = Object.keys(orgSettings);
+        if (!headersInOrgSettings.includes(headerName)) {
+          // If the current header name is not present in org settings,
+          // pick the details from user's settings and put it into merged settings object
+          set(mergedSettings, path, get(userSettings, path));
+          continue;
+        }
+        // If the current header name IS present in org settings, use it's scope information
+        set(mergedSettings, `${path}.scope`, get(orgSettings, `${path}.scope`));
+
+        for (let [tabName, tabDetails] of Object.entries(headerDetails.children)) {
+          // iterating on tabs inside headerName
+          path = `${headerName}.children`;
+          let tabsInOrgSettings = Object.keys(get(orgSettings, path));
+          if (!tabsInOrgSettings.includes(tabName)) {
+            // If the current tab name is not present in org settings,
+            // pick the details from user's settings and put it into merged settings object
+            set(
+              mergedSettings,
+              `${path}.${tabName}`,
+              get(userSettings, `${path}.${tabName}`)
+            );
+            continue;
+          }
+          // If the current tab name IS present in org settings, use it's scope information
+          set(
+            mergedSettings,
+            `${path}.${tabName}.scope`,
+            get(orgSettings, `${path}.${tabName}.scope`)
+          );
```

Comment:
lot of redundancy in code across the three levels no? - using functions will make this so much simpler to read and avoid redundancy

Reply:
Suggestions? Can't think of anything for this
Note - this has been update though

Reply:
recursion?

Reply:
I really can't think of how to apply recursion here
it's kind of becoming over complicated (for me at least)
If you can think of how to apply recursion here do let me know

Also I found no resources online to traverse a map recursively